Update "extractResourceId" function

anshumgargdd opened the pull request:
### What does this PR do?
This extends "extractResourceId" function to extract "source" tag from applications that are deployed at Tenant level.

The existing behavior is to discard all resourceId that don't begin with "/subscriptions/" thus it doesn't handle Tenant level resourceId that begin with "/tenants/".

Adds a else if block to handle resourceIds with Tenant level deployemnts(Like Active Directory)

I have kept the source and tag extraction logic similar to what currently exists.

A brief description of the change being made with this pull request.

### Motivation

Currently we don't get the "source" tag for Azure Active Directory thus making it not possible to ship logs integration pipeline and security rules for the integration. 

### Additional Notes

@steveharrington1 for additional context on the Azure integrations side.

### Checklist

- [ ] Member of the datadog team has run integration tests

anshumgargdd committed:
Update "extractResourceId" function

This extends "extractResourceId" function to extract "source" tag from applications that are deployed at Tenant level.

The existing behavior is to discard all resourceId that don't begin with "/subscriptions/" thus it doesn't handle Tenant level resourceId that begin with "/tenants/".

Adds a else if block to handle resourceIds with Tenant level deployemnts(Like Active Directory)

I have kept the source and tag extraction logic similar to what currently exists.

steveharrington1 commented on azure/activity_logs_monitoring/index.js:
```diff
@@ -135,22 +135,31 @@ function extractResourceId(record) {
+      var resourceId = record.resourceId.toLowerCase().split('/');
+      if (resourceId.length > 2 && resourceId[4]) {
+          metadata.tags.push('tenant:' + resourceId[2]);
+          metadata.source = resourceId[4].replace('microsoft.', 'azure.').replace('aadiam', 'active_directory');
```
@anshumgargdd we'll want to update this to remove the '_' like we talked about. This will make it match the format of all the other azure source.

Updatted format should be: 'source:azure.activedirectory'

anshumgargdd replied:
Done.

anshumgargdd committed:
Updating active_directory to activedirectory

Azure AD source changed from azure.active_directory to azure.activedirectory to match other azure source convention.

graybjones commented on azure/activity_logs_monitoring/index.js:
```diff
@@ -135,22 +135,31 @@ function extractResourceId(record) {
-        metadata.tags.push('resource_group:' + resourceId[4]);
+    else if(record.resourceId.toLowerCase().startsWith('/tenants/')){
+      var resourceId = record.resourceId.toLowerCase().split('/');
+      if (resourceId.length > 2 && resourceId[4]) {
```
Is there any chance the resourceId.length could be > 2 but less than 5?

anshumgargdd replied:
Unfortunately we don't have an exhaustive list of resourceIds so I am unsure. However in that case we just won't process the resourceId to extract source and assign 'azure' as the default source which is the behavior right now.

graybjones replied:
I think line 157 should read:

`if (resourceId.length > 4 && resourceId[4]) {`

That would cover the situation where the resourceId is greater than 2, but less than 5.

anshumgargdd replied:
Fixed.

graybjones reviewed (changes requested):
A small change at line 157 mentioned in the existing conversation. Thanks.

anshumgargdd committed:
Updating if condition for tenant

to (resourceId.length > 4 && resourceId[4])

graybjones reviewed (approved):
Thanks!
